tests/themispp/secure_session_test.hpp

```diff
@@ -113,19 +119,44 @@ namespace themispp{
 
       sput_fail_if(client.is_established(), "using moved session again", __LINE__);
     }
+
+    void secure_session_ownership(){
+      std::string client_id("client");
+
+      auto client_callbacks_shared = std::make_shared<callback>();
+      auto client_callbacks_unique = std::unique_ptr<callback>(new callback());
+
+      themispp::secure_session_t client_shared(std::vector<uint8_t>(client_id.c_str(), client_id.c_str()+client_id.length()), std::vector<uint8_t>(client_priv, client_priv+sizeof(client_priv)), std::move(client_callbacks_shared));
+      themispp::secure_session_t client_unique(std::vector<uint8_t>(client_id.c_str(), client_id.c_str()+client_id.length()), std::vector<uint8_t>(client_priv, client_priv+sizeof(client_priv)), std::move(client_callbacks_unique));
+
+      themispp::secure_session_t client_shared_moved = std::move(client_shared);
+      themispp::secure_session_t client_unique_moved = std::move(client_unique);
+
+      sput_fail_if(client_shared_moved.is_established(), "using shared session", __LINE__);
+      sput_fail_if(client_unique_moved.is_established(), "using unique session", __LINE__);
```

Comment:
will be good to check that `client_unique` is invalid after move. check bad case too)

Reply:
Done. I've added a check that secure session throws after being moved out. (This is true for `client_shared` as well. Ownership over the transport does not matter here.)